components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/resources/web/role/role-mgt.jsp

```diff
@@ -385,7 +431,92 @@
                             }
                         }
                     }
-               %>
+                %>
+
+                <%
+                    if (showDeletePendingRoles != null) {
+                        for (Iterator<FlaggedName> iterator = showDeletePendingRoles.iterator(); iterator.hasNext(); ) {
+                            FlaggedName flaggedName = iterator.next();
+                            if (flaggedName != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug in Axis!!
+                                if (CarbonConstants.REGISTRY_ANONNYMOUS_ROLE_NAME.equals(flaggedName.getItemName())) {
+                                    continue;
+                                }
+                                if (userRealmInfo.getAdminRole().equals(flaggedName.getItemName()) &&
+                                        !userRealmInfo.getAdminUser().equals(currentUser)) {
+                                    continue;
+                                }
+                                String roleName = CharacterEncoder.getSafeText(flaggedName.getItemName());
+                                String disPlayName = CharacterEncoder.getSafeText(flaggedName.getItemDisplayName());
+                                if (disPlayName == null) {
+                                    disPlayName = roleName;
+                                }
+                                String disPlayRoleName = disPlayName + " " + "[Pending Roles for Delete]";
```

Comment:
Concat the two Strings.
The suffix should be changed as "Pending to be deleted" ?